verilog/preprocessor/verilog_preprocess.cc

```diff
@@ -587,48 +598,19 @@ absl::Status VerilogPreprocess::HandleInclude(
   std::filesystem::path file_path =
       std::string(token_text.substr(1, token_text.size() - 2));
 
-  std::string file_path_to_search = file_path.string();
-
-  if (file_path.is_relative()) {
-    bool found = false;
-    for (const auto& base_dir : preprocess_info_.include_dirs) {
-      file_path_to_search =
-          verible::file::JoinPath(base_dir, file_path.string());
-      auto search_status = verible::file::FileExists(file_path_to_search);
-      if (search_status.ok()) {
-        found = true;
-        break;
-      }
-    }
-    if (!found) {
-      preprocess_data_.errors.push_back(
-          {**token_iter,
-           absl::StrCat(file_path_to_search, ": no such file found.")});
-      return absl::InvalidArgumentError("ERROR: included file can't be found.");
-    }
-  } else {
-    if (auto status = verible::file::FileExists(file_path_to_search);
-        !status.ok()) {
-      preprocess_data_.errors.push_back(
-          {**token_iter, std::string(status.message())});
-      return status;
-    }
-  }
-
-  // Actually open the file.
-  std::string source_contents;
-  if (auto status =
-          verible::file::GetContents(file_path_to_search, &source_contents);
-      !status.ok()) {
+  // Use the provided FileOpener to open the included file.
+  const auto status_or_file = file_opener_(file_path.string());
+  if (!status_or_file.ok()) {
     preprocess_data_.errors.push_back(
-        {**token_iter, std::string(status.message())});
-    return status;
+        {**token_iter, std::string(status_or_file.status().message())});
+    return status_or_file.status();
   }
+  const auto source_contents = *status_or_file;
 
   // Creating a new "VerilogPreprocess" object for the included file,
   // With the same configuration and preprocessing info (defines, incdirs) as
   // the main one.
-  verilog::VerilogPreprocess child_preprocessor(config_);
+  verilog::VerilogPreprocess child_preprocessor(config_, file_opener_);
   child_preprocessor.setPreprocessingInfo(preprocess_info_);
 
   // TODO(karimtera): limit number of nested includes, detect cycles? maybe.
```

Comment:
Below we should add a TODO that we we ideally get a MemBlock from the file opener, so that we don't have to create a second copy
(because the file opener now returns a string-view which then is copied again into another string in TextStructure).

TODO is ok for now, as I will have to refactor a little other stuff first before this is possible